app/Http/Controllers/Api/UserController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Api\UserRequest;
+use App\Http\Requests\Api\UserRoleRequest;
+use App\Models\User;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Hash;
+
+class UserController extends Controller
+{
+    public function index(): Response
+    {
+        return response(User::orderBy('name')->paginate(100));
+    }
+
+    public function update(UserRequest $request, User $user): Response
+    {
+        $validated = $request->validated();
+        $updatedPassword = Hash::make($validated['password']);
+        $user->update([
+            'name' => $validated['name'],
+            'email' => $validated['email'],
+            'password' => $updatedPassword,
+        ]);
+
+        return response($user);
+    }
+
+    public function updateUserRoles(UserRoleRequest $request, User $user): Response
+    {
+        $validated = $request->validated();
+        $roles = $validated['roles'];
+        $userWithUpdatedRoles = $user->syncRoles($roles);
+
+        return response($userWithUpdatedRoles);
```

Comment:
> same feedback here, I think we should the user object with updated set of roles.

yes, it's already returned the user with the new set of roles as a response, I think the name `isUpdated` suggests that it is returning a boolean value that's why I changed the variable name according to their return statement. `$userWithUpdatedRoles` is return the user object with the new set of roles.